File: src/ast/BracedRangeExpression.java
package ast;

import java.io.StringWriter;

public class BracedRangeExpression extends BaseNode {
    private BaseNode _firstNode;
    private BaseNode _lastNode;
    private BaseNode _expression;

    public BracedRangeExpression(BaseNode firstNode, BaseNode lastNode, BaseNode expression) {
        super(NodeType.BracedRangeExpression);
        _firstNode = firstNode;
        _lastNode = lastNode;
        _expression = expression;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("[");
        _firstNode.Print(writer);
        writer.write(" ... ");
        _lastNode.Print(writer);
        writer.write("]");

        if (!(_expression instanceof BracedExpression) || !(_expression instanceof BracedRangeExpression)) {
            writer.write(" = ");
        }

        _expression.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/BracedRangeExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class BracedRangeExpression(private val firstNode: BaseNode, private val lastNode: BaseNode, private val expression: BaseNode) : BaseNode(NodeType.BracedRangeExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("[")
        firstNode.print(writer)
        writer.write(" ... ")
        lastNode.print(writer)
        writer.write("]")
        writer.write(" = ")
        expression.print(writer)
    }

}